app/models/user.rb

```diff
@@ -16,6 +17,35 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  class << self
+    def digest string
+      if cost = ActiveModel::SecurePassword.min_cost
+        BCrypt::Engine::MIN_COST
+      else
+        BCrypt::Engine.cost
+      end
+      BCrypt::Password.create(string, cost: cost)
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    @remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  def authenticated? remember_token
+    return false unless remember_digest
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
```

Comment:
có những loại update nào nhỉ ? và chú nghĩ nên dùng loại nào

Reply:
Có 4 loại update_column, update_attribute, update và update_all. Như buổi review trước với chị dương thì cũng đã được khuyên dùng update, giúp update toàn bộ bản ghi (ngay cả những trường tự sinh như update_at, create_at) ạ!

<3